tests/ast/test_pre_parser.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+
+from vyper.ast.pre_parser import validate_version_pragma
+from vyper.exceptions import VersionException
+
+
+def test_validate_version_pragma_valid(monkeypatch):
+    _mock_version(monkeypatch)
+    validate_version_pragma(' @version 0.1.0b17', (1, 0))
+    validate_version_pragma(' @version 0.1.3', (1, 0))
+    assert True
+
+
+def test_validate_version_pragma_incompatible(monkeypatch):
+    _mock_version(monkeypatch)
+    with pytest.raises(VersionException):
+        validate_version_pragma(' @version 0.2.0b17', (1, 0))
+    with pytest.raises(VersionException):
+        validate_version_pragma(' @version 3.1.0b17', (1, 0))
+
+
+def test_validate_version_pragma_invalid(monkeypatch):
+    _mock_version(monkeypatch)
+    with pytest.raises(VersionException):
+        validate_version_pragma(' @version 0.1', (1, 0))
+
+
+def _mock_version(monkeypatch):
+    monkeypatch.setattr("vyper.__version__", "0.1.0b17")
```

Comment:
Can we move this to the top, and make it it's own fixture?
```suggestion
@pytest.fixture
def mock_version(monkeypatch):
    def set_version(version=vyper.__version__):
        monkeypatch.setattr("vyper.__version__", version)
    return set_version
```

You would import `mock_version` instead of `monkey_patch`, and it also allows setting different versions to check (or leaving it empty and checking against whatever the latest is)

Reply:
Roger that!